Jenkinsfile

```diff
@@ -1,35 +1,215 @@
+                    }
+                } catch(err) {
+                    utils.ircNotification(config, [stage: 'Unit Test Code Image', status: 'failure'])
+                    throw err
+                }
+            }
+        },
+        push: {
+            node {
+                unstash 'scripts'
+                try {
+                    utils.pushDockerhub('mozorg/bedrock_code')
+                } catch(err) {
+                    utils.ircNotification(config, [stage: 'Dockerhub Code Push', status: 'warning'])
+                }
+            }
+        }
     }
-    else if ( branch == 'prod') {
-        ircNotification('Stage & Prod Deploys', 'starting')
-        build 'bedrock_base_image'
+
+    stage ('Build L10n') {
+        node {
+            unstash 'scripts'
+            try {
+                utils.buildDockerImage(dockerfile: 'bedrock_l10n', fromDockerfile: 'bedrock_code', script: 'include_l10n.sh')
+            } catch(err) {
+                utils.ircNotification(config, [stage: 'L10n Build', status: 'failure'])
+                throw err
+            }
+            utils.ircNotification(config, [stage: 'Docker Builds', status: 'complete'])
+        }
     }
-    else if ( branch ==~ /^demo__[\w-]+$/ ) {
-        ircNotification('Demo Deploy', 'starting')
+
+    stage ('Push Images') {
+        // push images to docker hub and internal registries
+        // FIXME(pmac) this should be dynamic based on which regions to which
+        //             this branch will be deployed
+        parallel dockerhub: {
+            node {
+                unstash 'scripts'
+                try {
+                    utils.pushDockerhub('mozorg/bedrock_l10n', 'mozorg/bedrock')
+                } catch(err) {
+                    utils.ircNotification(config, [stage: 'Dockerhub Push Failed', status: 'warning'])
+                }
+            }
+        },
+        registry_usw: {
+            node {
+                unstash 'scripts'
+                try {
+                    utils.pushPrivateReg(config.regions.usw.registry_port)
+                } catch(err) {
+                    utils.ircNotification(config, [stage: 'US-West Registry Push', status: 'failure'])
+                    throw err
+                }
+            }
+        },
+        registry_euw: {
+            node {
+                unstash 'scripts'
+                try {
+                    utils.pushPrivateReg(config.regions.euw.registry_port)
+                } catch(err) {
+                    utils.ircNotification(config, [stage: 'EU-West Registry Push', status: 'failure'])
+                    throw err
+                }
+            }
+        }
+        node {
+            unstash 'scripts'
+            unstash 'tests'
+            // prep for next stage
+            sh 'docker/jenkins/build_integration_test_image.sh'
+            utils.ircNotification(config, [stage: 'Docker Image Pushes', status: 'complete'])
+        }
+    }
+
```

Comment:
Insert a `milestone()` right here to ensure that " that older builds of a job will not overwrite a newer build.", i.e. that we won't be pushing an old build to the server?

https://jenkins.io/blog/2016/10/16/stage-lock-milestone/